tests/components/viam/test_config_flow.py

```diff
@@ -0,0 +1,238 @@
+            CONF_SECRET: "randomSecreteForRobot",
+        },
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] is None
+    assert result["step_id"] == "robot"
+
+    _client, mock_robot = mock_viam_client
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {
+            CONF_ROBOT: mock_robot.id,
+        },
+    )
+    await hass.async_block_till_done()
+
+    assert result["type"] == FlowResultType.CREATE_ENTRY
+    assert result["title"] == "home"
+    assert result["data"] == {
+        CONF_ADDRESS: "my.robot.cloud",
+        CONF_SECRET: "randomSecreteForRobot",
+        CONF_ROBOT_ID: mock_robot.id,
+        CONF_CREDENTIAL_TYPE: CRED_TYPE_LOCATION_SECRET,
+    }
+
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+@patch(
+    "viam.app.viam_client.ViamClient.create_from_dial_options",
+    side_effect=CannotConnect,
+)
+async def test_form_missing_secret(
+    _mock_create_client: AsyncMock, hass: HomeAssistant
+) -> None:
+    """Test we handle cannot connect error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {
+            CONF_CREDENTIAL_TYPE: CRED_TYPE_API_KEY,
+        },
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == "auth_api_key"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {
+            CONF_API_ID: "someTestId",
+            CONF_API_KEY: "",
+        },
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == "auth_api_key"
+    assert result["errors"] == {"base": "cannot_connect"}
+
+
+@patch.object(ViamClient, "create_from_dial_options", return_value=None)
+async def test_form_cannot_connect(
+    _mock_create_client: AsyncMock,
+    hass: HomeAssistant,
+    mock_setup_entry: AsyncMock,
+) -> None:
+    """Test we handle cannot connect error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {
+            CONF_CREDENTIAL_TYPE: CRED_TYPE_API_KEY,
+        },
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == "auth_api_key"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {
+            CONF_API_ID: "someTestId",
+            CONF_API_KEY: "randomSecureAPIKey",
+        },
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == "auth_api_key"
+    assert result["errors"] == {"base": "cannot_connect"}
+
+
+@patch(
+    "viam.app.viam_client.ViamClient.create_from_dial_options", side_effect=Exception
+)
```

Comment:
Please only patch the library where we use them. You can use parametrize to make these tests nice and together.

We should repatch the library to work again, so we can verify the flow is able to recover

Reply:
I did my best to follow this feedback. I moved the patching to use in context where it was required.

I tried to parametrize the three tests together but couldn't get the patching to work as a parameter. I didn't see any other examples of this for component tests, so maybe you can point to a different reference for this pattern.